pickle.c

```diff
@@ -868,6 +868,26 @@ static void picolFreeArgList(pickle_t *i, const int argc, char **argv) {
 	picolFree(i, argv);
 }
 
+static int hexchartonibble(int c) {
+	c = tolower(c);
+	if ('a' <= c && c <= 'f')
+		return 0xa + c - 'a';
+	return c - '0';
+}
+
+/* converts up to two characters and returns number of characters converted */
+static int hexstr2toint(const char *str, int *const val) {
+	assert(str);
+	assert(val);
+	if (!isxdigit(*str))
```

Comment:
I prefer to always assign a value to my inputs before returning so everything is in a defined state.